client/src/components/Dashboard.tsx

```diff
@@ -0,0 +1,174 @@
+    // Retrieves data; parses body 
+    async function http<T>(
+        request: RequestInfo
+    ): Promise<HttpResponse<T>> {
+        const response: HttpResponse<T> = await fetch(request);
+        try {
+            response.parsedBody = await response.json();
+        } catch (ex) { }
+
+        if (!response.ok) {
+            throw new Error(response.statusText);
+        }
+        return response;
+    }
+
+    // Consumer 
+    async function getData(url: string, measurement: string, setter: Function) {
+        let response: HttpResponse<Data[]>;
+        try {
+            response = await http<Data[]>(url);
+            if (response.parsedBody != undefined) {
+                const body: Data[] = response.parsedBody;
+                if (measurement == "Engines") {
+                    engineTimeSeriesDataFormat(body, "engine-1", setEngine1Data);
+                    engineTimeSeriesDataFormat(body, "engine-2", setEngine2Data);
+                    engineTimeSeriesDataFormat(body, "engine-3", setEngine3Data);
+                }
+                else {
+                    timeSeriesDataFormat(body, measurement, setter);
+                }
+            }
+        } catch (response) {
+            console.log("Error: ", response);
+        }
+    }
+
+    // Formats the json data into a format readible by TimeSeries 
+    function timeSeriesDataFormat(jsonData: Data[], varName: string, setter: Function) {
+        let pointsRaw: any[];
+        pointsRaw = [];
+        for (let i = 0; i < jsonData.length; i++) {
+            const date = new Date(jsonData[i]["_time"]);
+            pointsRaw.push(new TimeEvent(date, { value: jsonData[i]["_value"] }));
+        }
+        const series = new TimeSeries({
+            name: varName,
+            columns: ["time", "value"],
+            events: pointsRaw
+        });
+        setter(series);
+        setLoading(false);
+    }
+
+    // Formats the engine json data 
+    function engineTimeSeriesDataFormat(jsonData: Data[], varName: string, setter: Function) {
+        let pointsRaw: any[];
+        pointsRaw = [];
+        for (let i = 0; i < jsonData.length; i++) {
+            const engine = jsonData[i]["_measurement"];
+            if (engine == varName) {
+                const date = new Date(jsonData[i]["_stop"]);
+                pointsRaw.push(new TimeEvent(date, { value: jsonData[i]["_value"] }));
+            }
+        }
+        const series = new TimeSeries({
+            name: varName,
+            columns: ["time", "value"],
+            events: pointsRaw
+        });
+        setter(series);
+        setLoading(false);
+    }
+
+    // Peforms all neecssary functions on all 3 endpoints 
+    function timeSeriesFormatAll() {
+        getData("http://localhost:8000/altitude", "Altitude", setAltitudeData);
+        getData("http://localhost:8000/acceleration", "Acceleration", setAccelerationData);
+        getData("http://localhost:8000/engine", "Engines", setEngine1Data);
```

Comment:
Smth you can do is factor out the base url of your api (and put it in a constant)